raiden/tests/unit/test_tokennetwork.py

```diff
@@ -550,3 +556,224 @@ def test_routing_updates(
+
+    # test routing with all nodes available
+    chain_state.nodeaddresses_to_networkstates = {
+        address1: NODE_NETWORK_REACHABLE,
+        address2: NODE_NETWORK_REACHABLE,
+        address3: NODE_NETWORK_REACHABLE,
+    }
+
+    route1 = get_best_routes(
+        chain_state=chain_state,
+        token_network_id=token_network_state.address,
+        from_address=our_address,
+        to_address=address1,
+        amount=50,
+        previous_address=None,
+    )
+    assert len(route1) == 2
+
+    route2 = get_best_routes(
+        chain_state=chain_state,
+        token_network_id=token_network_state.address,
+        from_address=our_address,
+        to_address=address1,
+        amount=51,
+        previous_address=None,
+    )
+    assert len(route2) == 1
+
+    # test routing with node 2 offline
+    chain_state.nodeaddresses_to_networkstates = {
+        address1: NODE_NETWORK_REACHABLE,
+        address2: NODE_NETWORK_UNREACHABLE,
+        address3: NODE_NETWORK_REACHABLE,
+    }
+
+    route1 = get_best_routes(
+        chain_state=chain_state,
+        token_network_id=token_network_state.address,
+        from_address=our_address,
+        to_address=address1,
+        amount=50,
+        previous_address=None,
+    )
+    assert len(route1) == 1
+
+    route2 = get_best_routes(
+        chain_state=chain_state,
+        token_network_id=token_network_state.address,
+        from_address=our_address,
+        to_address=address1,
+        amount=51,
+        previous_address=None,
+    )
+    assert len(route2) == 0
+
+    # test routing with node 3 offline
+    # the routing doesn't care as node 3 is not directly connected
+    chain_state.nodeaddresses_to_networkstates = {
+        address1: NODE_NETWORK_REACHABLE,
+        address2: NODE_NETWORK_REACHABLE,
+        address3: NODE_NETWORK_UNREACHABLE,
+    }
+
+    route1 = get_best_routes(
+        chain_state=chain_state,
+        token_network_id=token_network_state.address,
+        from_address=our_address,
+        to_address=address1,
+        amount=50,
+        previous_address=None,
+    )
+    assert len(route1) == 2
+
+    route2 = get_best_routes(
+        chain_state=chain_state,
+        token_network_id=token_network_state.address,
+        from_address=our_address,
+        to_address=address1,
+        amount=51,
+        previous_address=None,
+    )
+    assert len(route2) == 1
+
+    # test routing with node 1 offline
+    chain_state.nodeaddresses_to_networkstates = {
+        address1: NODE_NETWORK_UNREACHABLE,
+        address2: NODE_NETWORK_REACHABLE,
+        address3: NODE_NETWORK_REACHABLE,
+    }
+
+    route1 = get_best_routes(
+        chain_state=chain_state,
+        token_network_id=token_network_state.address,
+        from_address=our_address,
+        to_address=address1,
+        amount=50,
+        previous_address=None,
+    )
+    assert len(route1) == 1  # right now the channel to 2 gets filtered out as it is offline
```

Comment:
Here and in the other asserts could you add more specific checks for the routes and not depend only on the length?

Reply:
Sure